packages/sn-controls-react/src/fieldcontrols/Avatar/Avatar.tsx

```diff
@@ -219,49 +167,42 @@ export class Avatar<T extends GenericContent, K extends keyof T> extends Compone
       case 'new':
         return (
           <FormControl
-            className={className}
             style={styles.root as any}
-            key={name as string}
+            key={this.props.settings.Name}
             component={'fieldset' as 'div'}
-            required={required}>
-            <InputLabel shrink={true} htmlFor={name as string}>
-              {this.props['data-labelText']}
+            required={this.props.settings.Compulsory}>
+            <InputLabel shrink={true} htmlFor={this.props.settings.Name}>
+              {this.props.settings.DisplayName}
             </InputLabel>
             <List
               dense={true}
               style={this.state.fieldValue.length > 0 ? styles.listContainer : { ...styles.listContainer, width: 200 }}>
-              {itemTemplate ? (
-                itemTemplate(this.state.fieldValue)
-              ) : (
+              {
                 <DefaultAvatarTemplate
-                  repositoryUrl={this.props['data-repository'].configuration.repositoryUrl}
+                  repositoryUrl={this.props.repository!.configuration.repositoryUrl}
                   add={this.addItem}
                   actionName="new"
-                  readOnly={this.props.readOnly}
+                  readOnly={this.props.settings.ReadOnly}
+                  url={this.state.fieldValue}
                   remove={this.removeItem}
-                  renderIcon={this.props['data-renderIcon'] ? this.props['data-renderIcon'] : renderIconDefault}
+                  renderIcon={this.props.renderIcon ? this.props.renderIcon : renderIconDefault}
                 />
-              )}
+              }
             </List>
-            {this.props['data-hintText'] ? <FormHelperText>{this.props['data-hintText']}</FormHelperText> : null}
-            {this.props['data-errorText'] ? <FormHelperText>{this.props['data-errorText']}</FormHelperText> : null}
+            {this.props.settings.Description ? (
+              <FormHelperText>{this.props.settings.Description}</FormHelperText>
+            ) : null}
 
             <Dialog onClose={this.handleDialogClose} open={this.state.pickerIsOpen}>
               <div style={styles.dialog}>
                 <Typography variant="h5" gutterBottom={true}>
                   {AVATAR_PICKER_TITLE}
                 </Typography>
                 <AvatarPicker
-                  path={
-                    this.props['data-selectionRoot']
-                      ? this.props['data-selectionRoot'][0]
-                      : this.props['data-uploadFolderPath']
-                  }
-                  allowedTypes={this.props['data-allowedTypes']}
-                  repository={this.props['data-repository']}
+                  path={this.props.settings.SelectionRoots ? this.props.settings.SelectionRoots[0] : ''}
```

Comment:
There should be a default url instead of an empty string. In this case the uploadFolderPath gave the possibility to add a default path where the avatar image should be uploaded if there's no dedicated place for the avatars and the selection roots is empty. For example if I want to force users to upload their avatar into their own profile doclib or something like that which cannot be defined in the selectionroot

Reply:
I've put back uploadFolderPath to this control.